dev-java/hamcrest-core/hamcrest-core-1.3-r2.ebuild

```diff
@@ -37,24 +37,22 @@ PATCHES=(
 
 src_prepare() {
 	default
-	# Empty out the contents of the generator target; it has already been built.
-	eapply "${FILESDIR}/hamcrest-1.3-empty_generator.patch"
-
-	# Fix problems with Javadoc target.
-	eapply "${FILESDIR}/hamcrest-core-1.3-fix_javadoc.patch"
-
 	find -iname "*.jar" -exec rm -v {} + || die "Unable to clean bundled JAR files"
-
-	local cp="build/${P}.jar"
-	cp="${cp}:$(java-pkg_getjars --build-only --with-dependencies hamcrest-generator-${SLOT})"
-	EANT_EXTRA_ARGS="-Dversion=${PV} -Dgentoo.classpath=${cp}"
 }
 
-src_install() {
-	java-pkg_newjar build/${P}.jar ${PN}.jar
+src_compile() {
+	java-pkg-simple_src_compile
 
-	dodoc README.txt CHANGES.txt
+	"$(java-config -J)" \
+		-cp /usr/share/hamcrest-generator-1.3/lib/hamcrest-generator.jar:/usr/share/qdox-1.12/lib/qdox.jar:hamcrest-core.jar \
```

Comment:
this should be `-cp $(java-config --with-dependencies --classpath hamcrest-generator:1.3):${PN}.jar`. you do not want to hardcode the paths.

Reply:
This I need to compare and to check more in detail.  Thanks for the advise

Reply:
@fordfrog 
 There were our latest conversations crossing - Sorry for confusing number of commits